xuluomin/src/server/src/server.cpp

```diff
@@ -0,0 +1,77 @@
+#include <ros/ros.h>
+#include <time.h>
+
+#include <server/time.h>
+#include <server/show.h>
+
+#include <vector>
+#include <iostream>
+
+server::show srv;
+
+ros::NodeHandle *n_pointer;
+
+std::vector<ros::Subscriber> server_subscribers;
+
+ros::ServiceServer service_server;
+
+void time_callback(const server::time::ConstPtr& time)
+{
+    time_t  cur_t=time->second;
+    struct tm *t=localtime(&cur_t);
+
+    int year = t->tm_year + 1900; // years since 1900
+    int month = t->tm_mon + 1; // months since January - [0, 11]
+    int day = t->tm_mday;
+
+    int hour = t->tm_hour;
+    int minute = t->tm_min;
+    int second = t->tm_sec;
+
+    ROS_INFO("The name of client:%s   time: %d-%d-%d  %d : %d : %d  ", time->name.c_str(), year, month, day, hour, minute, second);
```

Comment:
**[cpplint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> Lines should be <= 80 characters long  [whitespace/line_length] [2]